packages/sdk/src/agent.ts

```diff
@@ -0,0 +1,160 @@
+import { AskarModule } from '@credo-ts/askar'
+import { Agent, DidsModule, KeyDerivationMethod, X509Module, type X509ModuleConfigOptions } from '@credo-ts/core'
+import {
+  ConnectionsModule,
+  CredentialsModule,
+  DidCommModule,
+  DiscoverFeaturesModule,
+  HttpOutboundTransport,
+  MediationRecipientModule,
+  MessagePickupModule,
+  OutOfBandModule,
+  ProofsModule,
+  WsOutboundTransport,
+} from '@credo-ts/didcomm'
+import { OpenId4VcHolderModule } from '@credo-ts/openid4vc'
+import { agentDependencies } from '@credo-ts/react-native'
+import { askar } from '@openwallet-foundation/askar-react-native'
+import { ParadymWalletMustBeDidCommAgentError, ParadymWalletMustBeOpenId4VcAgentError } from './error'
+import { type LogLevel, logger } from './logger'
+
+export type SetupAgentOptions = {
+  /**
+   *
+   * Unique identifier of your wallet storage
+   *
+   */
+  id: string
+
+  /**
+   *
+   * Key that will be used to securely open your wallet
+   *
+   * @security
+   * This property must dealt with in great care, such as receiving it from a biometrics check
+   *
+   */
+  key: string
+
+  /**
+   *
+   * The log level of the events that occur within the wallet
+   *
+   */
+  logLevel?: LogLevel
```

Comment:
This would require us to export the Credo log-level. Do we want that? We could also just do `: 'debug'  | 'info'`, etc..